apiary-ranger-metastore-plugin/src/main/java/com/expediagroup/apiary/extensions/rangerauth/policyproviders/ApiaryRangerAuthAllAccessPolicyProvider.java

```diff
@@ -0,0 +1,160 @@
+import org.apache.ranger.admin.client.RangerAdminRESTClient;
+import org.apache.ranger.plugin.model.RangerPolicy;
+import org.apache.ranger.plugin.util.GrantRevokeRequest;
+import org.apache.ranger.plugin.util.ServicePolicies;
+import org.apache.ranger.plugin.util.ServiceTags;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Class to return a list of Ranger authorization policies that permits all access.  If this is configured in
+ * ranger-hive-security.xml in the "ranger.plugin.hive.policy.source.impl" property, Ranger will allow all actions.
+ * <p>
+ * The intent of this class is to only configure it on the Apiary read-only metastore so that Ranger auditing can still
+ * be used without Ranger authorization policies.  We have to do it this way, because Ranger auditing is down in the
+ * guts of the policy evaluation code, so there isn't a cleaner way to audit without policies that I can find.
+ */
+
+public class ApiaryRangerAuthAllAccessPolicyProvider implements RangerAdminClient {
+  @VisibleForTesting
+  static final List<String> ACCESS_NAMES =
+      Arrays.asList("select", "update", "create", "drop", "alter", "index", "lock", "all", "read", "write");
+
+  @VisibleForTesting
+  static final List<String> POLICY_RESOURCES = Arrays.asList("database", "column", "table");
+
+  @VisibleForTesting
+  static final String ALL_USERS_GROUP = "public";
+
+  private static final Logger log = LoggerFactory.getLogger(ApiaryRangerAuthPreEventListener.class);
+
+  private ServicePolicies servicePolicies;
+  private RangerAdminRESTClient rangerAdminRESTClient;
+
+
+  public ApiaryRangerAuthAllAccessPolicyProvider() {
+    log.info("Creating instance of ApiaryRangerAuthAllAccessPolicyProvider for Apiary read-only metastore");
+    this.rangerAdminRESTClient = new RangerAdminRESTClient();
+  }
+
+  public ApiaryRangerAuthAllAccessPolicyProvider(RangerAdminRESTClient rangerAdminRESTClient) {
+    log.info("Creating instance of ApiaryRangerAuthAllAccessPolicyProvider for Apiary read-only metastore");
+    this.rangerAdminRESTClient = rangerAdminRESTClient;
+  }
+
+  @Override
+  public void init(String serviceName, String appId, String configPropertyPrefix) {
+    servicePolicies = new ServicePolicies();
+    servicePolicies.setServiceName(serviceName);
+    servicePolicies.setServiceId(1L);
+    servicePolicies.setPolicyVersion(1L);
+    servicePolicies.setPolicyUpdateTime(new Date());
+
+    List<RangerPolicy> policies = new ArrayList<>();
+    policies.add(getPolicy(1L, serviceName, POLICY_RESOURCES));
+    servicePolicies.setPolicies(policies);
+
+    rangerAdminRESTClient.init(serviceName, appId, configPropertyPrefix);
+    log.info("Successfully initialized ApiaryRangerAuthAllAccessPolicyProvider");
```

Comment:
Not sure we really need to log this?

Reply:
I will make the change to debug().  But I will say, if you've spent much time trying to debug in hindsight a deployment of Apiary Ranger (or any part of Apiary), the fact that nothing is logged at INFO level makes it basically impossible without rebuilding a docker image, pushing it to artifactory, changing a terraform repo, and then deploying that.  Doable, but a huge waste of time.  Unless you're in prod, in which case it's not doable.

INFO level messages logged during initialization of an ECS container don't really contribute much to noisy logs, but aid forensic debugging immensely.    I get the whole code- and log-purity thing, but sometimes the practicality needs of supporting a complex infrastructure argue for a more pragmatic approach.

Reply:
I hear you but.... my reply would be - if you're debugging you should edit the logging config file and set the level to DEBUG instead of INFO, that's what it's there for! ;P But yes, if they're only emitted once at start up then it's not such an issue and I'm OK with leaving them at INFO if that really helps. The ones that drive me nuts are the type that get logged once per *request* and make the logs an absolute pain to trawl through later.

Reply:
I'm not too fuzzed if we have on info log in there just show that everything is running ok. Just for the sanity check.

Reply:
If you know you are debugging, then I totally agree.  However, forensic debugging is another thing completely.  You never know when you're going to have to do it, and if the devs left you some bread crumbs in the logs to make your life a little easier, you are usually quite happy about it. :)